src/Common/src/System/Net/SafeCloseSocket.cs

```diff
@@ -170,6 +170,8 @@ internal void CloseAsIs()
                 Dispose();
                 if (innerSocket != null)
                 {
+                    TryWakeup(innerSocket);
```

Comment:
What is the "Wait until it's safe" below referring to?  I'm wondering if it's ok to do this here (or even required to do this here), or if it should move down to before the InnerReleaseHandle.